src/jsifier.js

```diff
@@ -266,14 +266,12 @@ function JSify(data, functionsOnly) {
         // Emit the body of a JS library function.
         var proxyingMode = LibraryManager.library[ident + '__proxy'];
         if (USE_PTHREADS && proxyingMode) {
-          var sig = LibraryManager.library[ident + '__sig'];
-          if (!sig) throw 'Missing function signature field "' + ident + '__sig"! (Using proxying mode requires specifying the signature of the function)';
-          sig = sig.replace(/f/g, 'i'); // TODO: Implement float signatures.
           if (proxyingMode !== 'sync' && proxyingMode !== 'async') {
             throw 'Invalid proxyingMode ' + ident + '__proxy: \'' + proxyingMode + '\' specified!';
           }
           var sync = proxyingMode === 'sync';
-          if (sig.length > 1) {
+          var hasArgs = LibraryManager.library[ident].length > 0;
+          if (hasArgs) {
```

Comment:
Could this `assert typeof LibraryManager.library[ident] === 'function'`? This seems error prone especially if some refactoring occurs - some of `LibraryManager.library[ident]`s are strings and other values, so this could theoretically be doing a `string.length` and not a `function.length` operation? lgtm there

Reply:
Assertion added, good idea.

Reply:
Perfect, lgtm